File: app/src/main/java/com/task/paginationlist/presentation/piclist/views/activities/interfaces/IPaginatorAdapter.java
package com.task.paginationlist.presentation.piclist.views.activities.interfaces;

import android.view.View;

import com.task.paginationlist.presentation.piclist.interfaces.OnPickClickListener;

public interface IPaginatorAdapter {

    void setSuccessful(boolean successful);

    int getRealItemCount();

    void setReload(View.OnClickListener listener);

    void setOnPicClick(OnPickClickListener onPicClick);

    void setFooterVisibility(boolean isAllLoaded);
}

File: app/src/main/java/com/task/paginationlist/presentation/piclist/views/activities/interfaces/IPaginatorAdapter.kt
package com.task.paginationlist.presentation.piclist.views.activities.interfaces

import android.view.View

import com.task.paginationlist.presentation.piclist.interfaces.OnPickClickListener

interface IPaginatorAdapter {

    val realItemCount: Int

    fun setSuccessful(successful: Boolean)

    fun setReload(listener: View.OnClickListener)

    fun setOnPicClick(onPicClick: OnPickClickListener)

    fun setFooterVisibility(isAllLoaded: Boolean)
}
